From 921722093048702f5e0ec3c0435f14117fcd9111 Mon Sep 17 00:00:00 2001
From: Marco Martin <notmart@gmail.com>
Date: Fri, 7 Oct 2016 13:00:44 +0200
Subject: [PATCH 18/19] delete containments upon activity deletion

Summary:
after the latest activities refactor, containments weren't
deleted anymore upon activity deletion.

Test Plan:
created and deleted activities while monitoring how appletsrc file
was updated

Reviewers: ivan, #plasma, davidedmundson

Reviewed By: #plasma, davidedmundson

Subscribers: davidedmundson, plasma-devel

Tags: #plasma

Differential Revision: https://phabricator.kde.org/D2967
---
 shell/shellcorona.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/shell/shellcorona.cpp b/shell/shellcorona.cpp
index 74f6ded..152ffeb 100644
--- a/shell/shellcorona.cpp
+++ b/shell/shellcorona.cpp
@@ -1463,6 +1463,11 @@ void ShellCorona::activityAdded(const QString &id)
 void ShellCorona::activityRemoved(const QString &id)
 {
     m_activityContainmentPlugins.remove(id);
+    if (m_desktopContainments.contains(id)) {
+        for (auto cont : m_desktopContainments.value(id)) {
+            cont->destroy();
+        }
+    }
 }
 
 void ShellCorona::insertActivity(const QString &id, const QString &plugin)
-- 
2.7.4

